src/QsCompiler/Transformations/MonomorphizationValidation.cs

```diff
@@ -94,10 +87,10 @@ private class TypeTransformation : TypeTransformation<TransformationState>
         {
             public TypeTransformation(SyntaxTreeTransformation<TransformationState> parent) : base(parent, TransformationOptions.NoRebuild) { }
 
-            public override QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation> OnTypeParameter(QsTypeParameter tp)
-            {
-                throw new Exception("Type Parameter types must be resolved");
-            }
+            public override QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation> OnTypeParameter(QsTypeParameter tp) =>
+                SharedState.IntrinsicCallableSet.Contains(tp.Origin) 
+                ? QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>.NewTypeParameter(tp)
+                : throw new Exception("Type Parameter types must be resolved");
```

Comment:
Isn't this being thrown with the generic `Length` still being in the syntax tree after monomorphization?

Reply:
I see that `Length` is intrinsic, which means it won't trigger this exception, which is good. It being intrinsic also means it probably shouldn't appear in the explicit exemption list in this same file.

Reply:
See reply on the other comment.